frontend/src/utils/d3/nodes.js

```diff
@@ -0,0 +1,311 @@
+/* eslint-disable indent, no-use-before-define, func-names, no-return-assign */
+import * as d3 from 'd3';
+import graphUtils from './graph';
+import dragUtils from './drag';
+import highlighter from './highlighter';
+
+const rectSize = {
+  w: 50,
+  h: 25,
+};
+
+const deleteRectOffset = {
+  x: -50,
+  y: -90,
+};
+const deleteTextOffset = {
+  x: -25,
+  y: -77,
+};
+const editRectOffset = {
+  x: 0,
+  y: -90,
+};
+const editTextOffset = {
+  x: 25,
+  y: -77,
+};
+
+/**
+ * Handle creation of nodes
+ * @param {obj} node - d3 node object
+ * @param {obj} args - object of node properties
+ */
+function enter(node, args) {
+  const {
+    nodeRadius,
+    colorMap,
+  } = args;
+  return node.append('g')
+    .attr('class', 'node')
+    .attr('id', (d) => d.id)
+    // create node circle
+    .call((nodeCircle) => nodeCircle.append('circle')
+      .attr('class', (d) => `nodeCircle node-${d.id}`)
+      .attr('r', nodeRadius)
+      .attr('fill', (d) => colorMap((d.category && d.category[0]) || 'unknown'))
+      .style('cursor', 'pointer')
+      .call((n) => n.append('title')
+        .text((d) => {
+          let title = d.id;
+          if (d.name) {
+            title += `: ${d.name}`;
+          }
+          return title;
+        })))
+    // create node label
+    .call((nodeLabel) => nodeLabel.append('text')
+      .attr('class', 'nodeLabel')
+      .style('pointer-events', 'none')
+      .attr('text-anchor', 'middle')
+      .style('font-weight', 600)
+      .attr('alignment-baseline', 'middle')
+      .text((d) => {
+        const { name } = d;
+        return name || 'Something';
+      })
+      .each(graphUtils.ellipsisOverflow))
+    // create delete button
+    .call((nodeDelete) => nodeDelete.append('rect')
+      .attr('transform', `translate(${deleteRectOffset.x},${deleteRectOffset.y})`)
+      .attr('width', rectSize.w)
+      .attr('height', rectSize.h)
+      .attr('stroke', 'black')
+      .attr('fill', 'white')
+      .style('display', 'none')
+      .attr('class', (d) => `${d.id} deleteRect`))
+    // add delete button label
+    .call((nodeDeleteLabel) => nodeDeleteLabel.append('text')
+      .attr('dx', deleteTextOffset.x)
+      .attr('dy', deleteTextOffset.y)
+      .style('pointer-events', 'none')
+      .attr('text-anchor', 'middle')
+      .attr('alignment-baseline', 'middle')
+      .attr('class', (d) => `${d.id} deleteLabel`)
+      .style('display', 'none')
+      .text('delete'))
```

Comment:
This delete and edit text isn't centered correctly. Also, any chance we could make the button look a little nicer? Making the button rounded with a small drop shadow should make it look more like a material UI button.

Reply:
Could you share a screenshot of the button text not being centered please? I'm not seeing that on my machine.

Reply:
Made them look cooler in 57b695c

Reply:
<img width="675" alt="Screen Shot 2021-05-24 at 8 37 11 AM" src="https://user-images.githubusercontent.com/1634866/119348752-48924480-bc6b-11eb-9914-21632625653a.png">

The text isn't vertically centered on my computer. It's a pretty small issue so no worries if it's not an easy fix.

Reply:
This was interesting. It depended on the browser (firefox was misaligned, but chrome was fine), as well as old documentation for a previous d3 version. Fixed in d940bfceea04bf0f2c946ddaf6bee93b2f9ddf06

Reply:
Fix doesn't seem to have worked unfortunately, still looks the same on Firefox 88.

Reply:
That's cause I had only fixed the node buttons. Added edges in 8ad071bbb1addb707ee8db3588324fbf2792adc6